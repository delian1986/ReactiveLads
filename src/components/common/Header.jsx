import { NavLink as Link } from "react-router-dom";
import Logout from "../../containers/Logout";
import PropTypes from "prop-types";

export const Header = (props) => {
  const { email, isLoggedIn, photoUrl } = props;

  return (
    <nav className="navbar navbar-expand-md navbar-light bg-light">
      <Link className="navbar-brand mr-auto" to="/">
        ReactiveLads
      </Link>
      {isLoggedIn ? (
        <>
          <li className="nav-link">
            <a href="#" className="nav-link disabled">
              Hello, {email} !
            </a>
          </li>
          <img
            className="circle-round"
            style={{
              height: "40px",
              borderRadius: "50%"
            }}
            src={photoUrl && photoUrl !== "" ? photoUrl : "./avatar_placeholder.png"}
            alt="user-photo"
          />
          <Link name="userDetails" className="nav-link" to="/userDetails">
            User Details
          </Link>
          <Logout />
        </>
      ) : (
        <>
          <Link className="nav-link" to="/login">
            Login
          </Link>
          <Link className="nav-link" to="/signUp">
            Register
          </Link>
        </>
      )}
    </nav>
  );
};

Header.propTypes = {
  email: PropTypes.string,
  isLoggedIn: PropTypes.bool
};
