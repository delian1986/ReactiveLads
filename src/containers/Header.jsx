import { connect } from "react-redux";
import { Header } from "../components/common/Header";
import { getEmail, getLoggedInStatus, getPhotoUrl } from "../store/selectors";

function mapStateToProps(state) {
  return {
    email: getEmail(state),
    isLoggedIn: getLoggedInStatus(state),
    photoUrl: getPhotoUrl(state)
  };
}

export default connect(mapStateToProps)(Header);
