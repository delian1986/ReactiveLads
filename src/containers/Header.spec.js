import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Header from "./Header";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";

describe("Header container tests", () => {
  describe("Logged in user", () => {
    const mockStore = configureStore([]);
    let store;
    let component;
    beforeEach(() => {
      store = mockStore({
        auth: {
          email: "test@test.bg",
          isLoggedIn: true
        }
      });
      store.dispatch = jest.fn();
      component = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>
      );
    });

    it("should match logged in snapshot", function () {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  describe("Anonymous user", () => {
    const mockStore = configureStore([]);
    let store;
    let component;
    beforeEach(() => {
      store = mockStore({
        auth: {
          isLoggedIn: false
        }
      });
      store.dispatch = jest.fn();
      component = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>
      );
    });

    it("should render <Header/> component without crashing", () => {
      renderer.act(() => {
        expect(component.root.findAllByType("a")).toHaveLength(3);
      });
    });
  });
});
