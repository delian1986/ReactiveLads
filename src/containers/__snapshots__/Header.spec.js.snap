// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container tests Logged in user should match logged in snapshot 1`] = `
<nav
  className="navbar navbar-expand-md navbar-light bg-light"
>
  <a
    aria-current="page"
    className="navbar-brand mr-auto active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    ReactiveLads
  </a>
  <li
    className="nav-link"
  >
    <a
      className="nav-link disabled"
      href="#"
    >
      Hello, 
      test@test.bg
       !
    </a>
  </li>
  <img
    alt="user-photo"
    className="circle-round"
    style={
      Object {
        "borderRadius": "50%",
        "height": "40px",
      }
    }
  />
  <a
    aria-current={null}
    className="nav-link"
    href="/userDetails"
    onClick={[Function]}
  >
    User Details
  </a>
  <button
    className="btn btn-link nav-link"
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</nav>
`;
